src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -1958,23 +1974,24 @@ static int do_list_mapped_devices(const std::string &format, bool pretty_format)
   Config cfg;
   NBDListIterator it;
   while (it.get(&cfg)) {
+    std::string snap = (cfg.snapid != CEPH_NOSNAP) ? "@" + std::to_string(cfg.snapid) :
+                                                     cfg.snapname;
     if (f) {
       f->open_object_section("device");
       f->dump_int("id", cfg.pid);
       f->dump_string("pool", cfg.poolname);
       f->dump_string("namespace", cfg.nsname);
       f->dump_string("image", cfg.imgname);
-      f->dump_string("snap", cfg.snapname);
+      f->dump_string("snap", snap);
       f->dump_string("device", cfg.devpath);
       f->dump_string("cookie", cfg.cookie);
       f->close_section();
     } else {
       should_print = true;
-      if (cfg.snapname.empty()) {
-        cfg.snapname = "-";
-      }
+      if (snap.empty())
+        snap = "-";
       tbl << cfg.pid << cfg.poolname << cfg.nsname << cfg.imgname
-          << cfg.snapname << cfg.devpath << cfg.cookie << TextTable::endrow;
+          << snap << cfg.devpath << cfg.cookie << TextTable::endrow;
```

Comment:
Nit: instead of reassigning snap above, I would do:
```suggestion
          << (snap.empty() ? "-" : snap) << cfg.devpath << cfg.cookie
          << TextTable::endrow;
```